From: Gerald Schaefer <geraldsc@de.ibm.com>
Subject: vmalloc: ipl failure with enabled memory cgroups
References: bnc#594473
Patch-mainline: Upstream submission made

Symptom:     Systems with large memory sizes and enabled memory cgroups
             fail to ipl
Problem:     The memory cgroups code allocates 10 megabytes from the
             vmalloc area for each gigabyte of memory. With the default
             virtual vmalloc size of 1GB the ipl fails for memory sizes
             bigger than 100 gigabyte.
Solution:    Increase the default virtual vmalloc size to 128 gigabyte.

Acked-by: John Jolly <jjolly@suse.de>

---
 arch/s390/include/asm/pgtable.h |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/arch/s390/include/asm/pgtable.h
+++ b/arch/s390/include/asm/pgtable.h
@@ -120,8 +120,8 @@ extern unsigned long VMALLOC_START;
 #define VMALLOC_END	0x7e000000UL
 #define VMEM_MAP_END	0x80000000UL
 #else /* __s390x__ */
-#define VMALLOC_SIZE	(1UL << 30)
-#define VMALLOC_END	0x3e040000000UL
+#define VMALLOC_SIZE	(128UL << 30)
+#define VMALLOC_END	0x3e000000000UL
 #define VMEM_MAP_END	0x40000000000UL
 #endif /* __s390x__ */
 
